### PR TITLE
Update Backup and ClusterRole constructors and to_dict function

### PR DESCRIPTION
**Pull request by oharan2**

### CODE REVIEW

File: ocp_resources/cluster_role.py [hunk 1]
```diff
@@ -10,42 +9,23 @@ class ClusterRole(Resource):
...
-        )
+        super().__init__(**kwargs)
         self.rules = rules
 
     def to_dict(self):
-        if not self.rules and not self.yaml_file:
-            raise ValueError("must send rules or yaml_file")
-        if not self.res:
-            super().to_dict()
+        super().to_dict()
```
Comment (oharan2) [inline ocp_resources/cluster_role.py:24]: @myakove why `super().to_dict()` both here and also in line 29? by mistake?

File: ocp_resources/backup.py [hunk 2]
```diff
@@ -10,30 +10,18 @@ class Backup(NamespacedResource):
...
-        yaml_file=None,
-        excluded_resources=None,
-        **kwargs,
-    ):
+    def __init__(self, included_namespaces=None, excluded_resources=None, **kwargs):
+        """
+        Args:
+            included_namespaces (list): Namespaces to include in the backup (optional).
+                If unspecified, all namespaces are included.
+            excluded_resources (list): Resources to exclude from the backup (optional).
```
Comment (rnetser) [inline ocp_resources/backup.py:18]:
```suggestion
            excluded_resources (list, optional): Resources to exclude from the backup.
```

File: ocp_resources/backup.py [hunk 3]
```diff
@@ -10,30 +10,18 @@ class Backup(NamespacedResource):
 
     api_group = NamespacedResource.ApiGroup.VELERO_IO
```
Comment (rnetser) [inline ocp_resources/backup.py:12]: please a link to the api

File: ocp_resources/cluster_role.py [hunk 4]
```diff
@@ -10,42 +9,23 @@ class ClusterRole(Resource):
...
-        self,
-        name=None,
-        client=None,
-        rules=None,
-        teardown=True,
-        yaml_file=None,
-        delete_timeout=TIMEOUT_4MINUTES,
-        **kwargs,
-    ):
+    def __init__(self, rules=None, **kwargs):
```
Comment (rnetser) [inline ocp_resources/cluster_role.py:12]: please add link to api

File: ocp_resources/cluster_role.py [hunk 5]
```diff
@@ -10,42 +9,23 @@ class ClusterRole(Resource):
...
-        if not self.res:
-            super().to_dict()
+        super().to_dict()
         if not self.yaml_file:
-            self.res["rules"] = self.rules
+            if not self.rules and not self.yaml_file:
+                raise ValueError("must send rules or yaml_file")
+            if not self.res:
+                super().to_dict()
+            if not self.yaml_file:
```
Comment (rnetser) [inline ocp_resources/cluster_role.py:30]: not needed, already under this `if` condition

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/backup.py (hunk 3 above)
Comment (oharan2) [reply ocp_resources/backup.py:14]: Done

File: ocp_resources/cluster_role.py (hunk 1 above)
Comment (myakove) [reply ocp_resources/cluster_role.py:26]: yes, should be only one

File: ocp_resources/cluster_role.py [hunk 6]
```diff
@@ -1,51 +1,31 @@
...
-        self,
-        name=None,
-        client=None,
-        rules=None,
-        teardown=True,
-        yaml_file=None,
-        delete_timeout=TIMEOUT_4MINUTES,
-        **kwargs,
-    ):
+    def __init__(self, rules=None, **kwargs):
```
Comment (omrirh) [inline ocp_resources/cluster_role.py:14]: Might have missed but why did we remove all init args? (e.g. yaml_file, name, client etc.)

Comment (oharan2) [reply ocp_resources/cluster_role.py:14]: They'll be passed as kwargs and we're passing them forward in `super().__init__(**kwargs)` (line 26)

File: ocp_resources/backup.py [hunk 7]
```diff
@@ -6,47 +6,38 @@
...
-    ):
+    def __init__(self, included_namespaces=None, excluded_resources=None, **kwargs):
+        """
+        Args:
+            included_namespaces (list, optional): Namespaces to include in the backup.
+                If unspecified, all namespaces are included.
+            excluded_resources (list, optional): Resources to exclude from the backup.
+                Resources may be shortcuts (e.g. 'po' for 'pods') or fully-qualified.
+        """
         if not included_namespaces:
```
Comment (rnetser) [inline ocp_resources/backup.py:23]: `included_namespaces` is optional; please remove.

File: ocp_resources/cluster_role.py [hunk 8]
```diff
@@ -1,51 +1,31 @@
 # -*- coding: utf-8 -*-
-from ocp_resources.constants import TIMEOUT_4MINUTES
 from ocp_resources.resource import Resource
 
 
 class ClusterRole(Resource):
     """
     ClusterRole object
```
Comment (rnetser) [inline ocp_resources/cluster_role.py:7]: please remove this line

File: ocp_resources/cluster_role.py [hunk 9]
```diff
@@ -1,51 +1,31 @@
...
+    'rbac_authorization_k8s_io' API official docs:
+        https://docs.openshift.com/container-platform/3.11/rest_api/rbac_authorization_k8s_io/rbac-authorization-k8s-io-index.html
```
Comment (rnetser) [inline ocp_resources/cluster_role.py:9]: please update to the api - https://kubernetes.io/docs/reference/kubernetes-api/authorization-resources/cluster-role-v1/

File: ocp_resources/cluster_role.py [hunk 10]
```diff
@@ -1,51 +1,31 @@
...
     def to_dict(self):
-        if not self.rules and not self.yaml_file:
-            raise ValueError("must send rules or yaml_file")
-        if not self.res:
-            super().to_dict()
         if not self.yaml_file:
+            if not self.rules:
+                raise ValueError("must send rules or yaml_file")
+            if not self.res:
+                super().to_dict()
```
Comment (rnetser) [inline ocp_resources/cluster_role.py:30]: should be first thing to call in `to_dict` without the `if`

**Review by rnetser — CHANGES_REQUESTED** [review]

**Review by myakove — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Review by myakove — DISMISSED** [review]

**Review by rnetser — APPROVED** [review]

**Pull request merged**